custom_scripts/py_scripts/db_scanner/db_scanner.py

```diff
@@ -0,0 +1,558 @@
+            if "_id" in str(document[key]):
+                continue
+            result_dict.update({key: str(document[key])})
+    return result_dict
+
+
+# TODO: improve regulars
+class MongoDB:
+    def __init__(self, ip: str):
+        self.ip = ip
+        self.port = 27017
+        self.server_timeout = 10000  # serverSelectionTimeoutMS
+        self.socket_timeout = 300000  # socketTimeoutMS
+        self.connection_timeout = 10000  # connectTimeoutMs
+        self.start = perf_counter()
+
+    def check_time_of_connection(self) -> None:
+        """
+        Check time of a connection
+        :return: None
+        """
+        if (
+            perf_counter() - self.start
+        ) > DBScannerDefaultValues.MAX_TIME_OF_CONNECTION_IN_SECONDS:
+            raise DBScannerConnectionTimeout(
+                "Connection was over because there are too many documents in db"
+            )
+
+    @staticmethod
+    def get_collection_names(client: database) -> (list or None, Exception or None):
+        """
+        Det collection name and catch all errors
+        :param client: MongoClient.database
+        :return: names of collections or error
+        """
+        try:
+            collection_names = client.list_collection_names()
+        except (AutoReconnect, OperationFailure, KeyError, InvalidName) as err:
+            print(f"Connection was over by server: {err}")
+            return None, err
+        return collection_names, None
+
+    @staticmethod
+    def get_number_of_documents(
+        client: database, collection_name: str
+    ) -> int or Exception:
+        """
+        Get number of documents in a collection
+        :param client: MongoClient.database
+        :param collection_name: name of a collection in database
+        :return: number of documents in a collection or exception
+        """
+        try:
+            docs_num = client[collection_name].count_documents({})
+        except (
+            AutoReconnect,
+            InvalidName,
+            OperationFailure,
+        ) as err:
+            print(f"Connection was over by server: {err}")
+            return err
+        return docs_num
+
+    def find_in_collection_with_reg(
+        self, client: database, collection_name: str
+    ) -> (dict or None, Exception or None):
+        """
+        Find values in collection using regular expressions for searching
+        :param client: MongoClient.database
+        :param collection_name: name of a collection in database
+        :return: founded result
+        """
+        resulted_value = self.get_number_of_documents(client, collection_name)
+        if not isinstance(resulted_value, int):
+            return None, resulted_value
+        docs_num = resulted_value
+        result_list = []
+        if not docs_num:
+            return None, None
+        keys = client[collection_name].find_one().keys()
+        keys_to_take = []
+        regulars = []
+        projection = {}
+        for key in keys:
```

Comment:
Is it applicable here?
https://docs.mongodb.com/manual/reference/operator/query/in/